io/pci/dlpar.py

```diff
@@ -153,10 +153,13 @@ def install_packages(self):
         Install required packages
         '''
         smm = SoftwareManager()
+        packages = ['ksh', 'src', 'rsct.basic', 'rsct.core.utils',
+                    'rsct.core', 'DynamicRM']
         detected_distro = distro.detect()
         self.log.info("Test is running on: %s", detected_distro.name)
-        if not smm.check_installed("ksh") and not smm.install("ksh"):
-            self.cancel('ksh is needed for the test to be run')
+        for pkg in packages:
+            if not smm.check_installed(pkg) and not smm.install(pkg):
+                self.cancel('%s is needed for the test to be run' % pkg)
         if detected_distro.name == "Ubuntu":
             if not smm.check_installed("python-paramiko") and not \
```

Comment:
just a one minor comment can this check also clubbed with https://github.com/avocado-framework-tests/avocado-misc-tests/pull/1094/files#diff-f5d7c870f9fca9cbe842de08c146d2e8R160

Reply:
I think this is specific to the distro i,e Ubuntu and the one I did is  common for all.

Reply:
yes i know , if distro == Ubuntu append package list  as you are already running for loop for all other package can be like 
```
packages = ['ksh', 'src', 'rsct.basic', 'rsct.core.utils',
                    'rsct.core', 'DynamicRM']
if detected_distro.name == "Ubuntu":
   packages.extend(['python-paramiko'])
for pkg in packages:
   if not smm.check_installed(pkg) and not smm.install(pkg):
                self.cancel('%s is needed for the test to be run' % pkg)
```

Reply:
Got it. will add that.